### ISSUE TITLE
AVT 1 - Form Item: DAP violations related to label

### ISSUE THREAD

**Issue by snidersd** [IBM/carbon-components-react#2188]
Environment
macOS Mojave version 10.14.2
Chrome version 72.0.3626.96 (Official Build) (64-bit)

Detailed Description
Run DAP with Feb. 2019 ruleset on the Form Item Component
The following violations are reported:
1. Each form control must have an associated label 
[H44: Using label elements to associate text labels with form controls](https://www.w3.org/TR/WCAG20-TECHS/H44)
[ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/TR/WCAG20-TECHS/ARIA16.html)
2. Provide descriptive text in label element
4.1.2 Name, Role, Value

Related to #1932

**Comment by paschalidi** [comment]
@snidersd thanks for reporting this! 
I appreciate that you took the time to look into this thoroughly! 
I am wondering before I start developing this if I get the requirements right. I hope you dont mind explaining to me a little bit more what is to be expected from 1. and 2.

> 1. Each form control must have an associated label

regarding 
> H44: Using label elements to associate text labels with form controls

Please help me to fully understand this before starting implementing! :) 
The `input` item has a label associated with i,  is that label missing some requirements or what one would expect this to have to be according to a11y standards? 

>ARIA16: Using aria-labelledby to provide a name for user interface controls

I think I get this, `input` elemtn needs to have `aria-labelledby`. Correct me if I am wrong! :) 


> 2. Provide descriptive text in label element
> 4.1.2 Name, Role, Value

Also I think this is straightforward enough, always feel free to correct me when I get something wrong. 
`label` element needs correct Name, Role, and Value. 

Can you please clarify a little bit more what does `Name` and `Value` would be by giving a simple example?


T h a n k s @snidersd !

**Comment by snidersd** [comment]
@paschalidi I just ran DAP again this morning and the issue is no longer present so I'm marking this one as closed.  The only thing DAP is reporting now is a false positive due to a known DAP defect with disabled fields. 
<img width="938" alt="Screen Shot 2019-04-22 at 8 43 21 AM" src="https://user-images.githubusercontent.com/21676914/56503327-ba22c600-64da-11e9-9fa5-19a4924075ce.png">

**Comment by snidersd** [comment]
@paschalidi  Sorry, I closed this issue in error. I retested the Form component rather then the Form item (see the correct screenshot below). 
<img width="1156" alt="Screen Shot 2019-04-22 at 8 50 55 AM" src="https://user-images.githubusercontent.com/21676914/56503679-ceb38e00-64db-11e9-8a1d-42c7130cddab.png">
